invenio_jobs/utils.py

```diff
@@ -44,3 +45,11 @@ def walk_values(obj, transform_fn):
             walk_values(val, transform_fn)
         else:
             obj[key] = transform_fn(val)
+
+
+def job_arg_json_dumper(obj):
+    """Handle non-serializable values such as datetimes when dumping the arguments of a job run."""
+    if isinstance(obj, datetime):
+        return obj.isoformat()
```

Comment:
As discussed, let's ensure here that there is no timezone or `Z` char added to the string, so that we avoid manipulating the string in any celery task or UI. Can you please add a test?